ultravox/data/dataset_config.py

```diff
@@ -12,6 +12,10 @@ class DataDictConfig(BaseModel):
     splits: List[str] = dataclasses.field(default_factory=list)
     num_samples: Optional[int] = None
     total_samples: int = 1
+    # epochs mode:
+    #  Weight is the number of copies of the dataset
+    # max steps mode:
+    #  Weight of the dataset is used to calculate the proporition of the total samples that comes from this dataset
```

Comment:
This distinction is very confusing and unexpected to any outsider. Using epochs or max_steps should not automatically change the behaviour other than changing the number of steps performed.

Reply:
Added dataset_multiplier

Reply:
Thanks Pat.

Though I believe the same thing is still happening at the end of the day: the distribution of data changes when you use epoch mode or not. To add to that, now there are a set of parameters that in one scenario don't do anything but they do something in the other.

Just to make it concrete, the simplest way I see to improve on that is:
1. Have a first-class CLI argument called something like `use_proportional_dataset_multiplier` instead. I don't think this has to be tied to whether `max_steps` is 0 or not. The source of confusion is this entanglement IMO, not the name weight vs data_multiplier.
2. I believe we can remove `weight` in favour of `dataset_multiplier`. Having both and then non-transparently switching between the two is going to be a source of confusion. The reason why I think you can get away with this is that I think `dataset_multiplier` is ambiguous enough that people will not assume they know what you mean. When they do and/or when they search for `dataset_multiplier`, they will see that a `use_proportional_dataset_multiplier` arguments exists.

The naming and everything I proposed are getting a bit too long, so feel free to adjust as you see fit.

Reply:
I agree with Farzad that this distinction is confusing. I think `weight` should just mean `dataset_multiplier` regardless of whether we `num_epochs` or `max_steps` to determine the number of updates. `weight` had a different meaning when it was first introduced because at that time we don't have a way to measure the length of a dataset.

Reply:
Ok, so I'll remove weight in favor of the dataset_multiplier, but that means we're removing the existing "proportional weight" functionality and if we're all fine with that I'll continue with it.

Reply:
And I'll call it dataset_multiplier re farzad's point that 
"is that if you divide all "weights" by a constant, nobody expects anything to change, but in this case it does."